clusterspec/clusterspec.go

```diff
@@ -0,0 +1,59 @@
+package clusterspec
+
+import (
+	"net/url"
+	"strings"
+
+	v1alpha1core "github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	v1alpha1provider "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/certs"
+	"github.com/giantswarm/microerror"
+)
+
+// Factory type acts as a state holder for supplemental v1alphaprovider.Cluster
+// info.
+type Factory struct {
+	baseCluster *v1alpha1provider.Cluster
```

Comment:
Do I understand correctly `baseCluser` is an object filled with common information coming from flags?

Reply:
More or less yes. The information comes from runtime configuration (parsed by Viper) and `flags` define structure for the configuration file. At least this is how I've understood this.

As can be seen in the factory function, this `baseCluster` is then copied and filled with "request" (request to create new guest cluster, a custom object in operator case) information.